fields/types/list/ListField.js

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable react/jsx-no-bind */
+
+import assign from 'object-assign';
+import React from 'react';
+import Field from '../Field';
+import Domify from 'react-domify';
+
+import { Fields } from 'FieldTypes';
+import { Button, GlyphButton } from '../../../admin/client/App/elemental';
+import InvalidFieldType from '../../../admin/client/App/shared/InvalidFieldType';
+
+function generateId () {
+	return Math.random().toString(36).substr(2, 9);
+};
+
+module.exports = Field.create({
+	displayName: 'ListField',
+	statics: {
+		type: 'List',
+	},
+	propTypes: {
+		fields: React.PropTypes.object.isRequired,
+		label: React.PropTypes.string,
+		onChange: React.PropTypes.func.isRequired,
+		path: React.PropTypes.string.isRequired,
+		value: React.PropTypes.array,
+	},
+	addItem () {
+		const value = this.props.value.slice();
+		value.push({
+			_id: generateId(),
+			_isNew: true,
+		});
+		this.props.onChange({
+			path: this.props.path,
+			value: value,
+		});
+	},
+	removeItem (index) {
+		const oldValue = this.props.value;
+		const value = oldValue.slice(0, index).concat(oldValue.slice(index + 1));
+		this.props.onChange({
+			path: this.props.path,
+			value: value,
+		});
+	},
+	handleFieldChange (index, event) {
+		const value = this.props.value.slice();
+		const update = value[index];
+		update[event.path] = event.value;
```

Comment:
This should work just fine for any level of nesting, unless I've missed something? Each `List` is self-contained, doesn't know what field types are in it (which could be more list field types)

You're right, mutation slipped past me. Should fix that.

I'm 👍 for moving this state into redux, let's do it after (at least) we release the beta. It'll touch all the field types and I don't know for sure there aren't dragons.